versioning/BRM/extentmap.cpp

```diff
@@ -1253,21 +1253,30 @@ void ExtentMap::loadVersion4(IDBDataFile* in)
         growEMShmseg(nrows);
     }
 
-    for (int i = 0; i < emNumElements; i++)
+    size_t progress = 0, writeSize = emNumElements * sizeof(EMEntry);
+    int err;
+    char *writePos = (char *) fExtentMap;
+    while (progress < writeSize)
```

Comment:
Could there ever be a case, such as corrupted file, where this condition is never met, causing an infinite loop?

Reply:
Right now there's not much protection against corruption, but if it get an early EOF or a read error, the if stmt below will catch it.  Garbled data is something it can't handle ATM.  If we want, we could add a checksum to the header w/o too much pain though, or checksums throughout, etc.  Easy to change the file format for these.